src/main/java/frc/robot/subsystems/Carousel.java

```diff
@@ -12,29 +12,59 @@
 import com.revrobotics.ColorSensorV3;
 import edu.wpi.first.wpilibj.Encoder;
 import edu.wpi.first.wpilibj.I2C;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 import frc.robot.Constants;
+import frc.robot.Robot;
 
 public class Carousel extends SubsystemBase {
     WPI_TalonSRX spinner;
     Encoder carouselEncoder;
     ColorSensorV3 colorSensor = new ColorSensorV3(I2C.Port.kMXP);
     int ballCount = 0;
-
+    // TODO: remove when removing old carousel command
+    public double currentCheckpoint = 0;
+    public double lastCheckpoint = 0;
+    double lastSpinSpeed;
+    public boolean backwards;
+    double lastBackTime;
 
     public Carousel() {
         spinner = new WPI_TalonSRX(Constants.CAROUSEL_CAN_ID);
         spinner.setNeutralMode(NeutralMode.Brake);
         carouselEncoder = new Encoder(7, 8);
+        backwards = false;
     }
 
     @Override
     public void periodic() {
         // This method will be called once per scheduler run
+        SmartDashboard.putNumber("Carousel/Color Sensor distance", getColorSensorProximity());
+        SmartDashboard.putNumber("Carousel/Output current", getCurrent());
+        SmartDashboard.putNumber("Carousel/Carousel ticks", -getTicks());
+
+        // First check is to see if the current is spiking, this would indicate a stuck ball
+        if (getCurrent() > 10.35) { 
+            lastBackTime = Robot.time(); // start timer for going backwards
+            backwards = true; // now we goin backwards
+        }
+        
+        if (backwards) {
+            // set the spinner going backwards to unjam the ball
+            // do NOT use the public spin() method, since we do not want this speed remembered
+            spinner.set(ControlMode.PercentOutput, -0.6);
```

Comment:
Oops. We should not set the spinner going backward, and then (if time expires) immediately set it forward. This should be in an else clause.

Reply:
I think I found a better way to write that section, let me know what you think of it

Reply:
Looks good to me.

Reply:
Yes, looks fine.